### PR TITLE
Add content translation demo

### PR DESCRIPTION
**Pull request by VP-DS**
COM-201, COM-198 

<details>
    <summary>Videos</summary>

# Example

https://github.com/vivid-planet/comet/assets/118435139/61776346-338e-4a9d-b254-c58516ac62fb

</details>

### CODE REVIEW

File: packages/api/cms-api/src/translation/translation.module.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+import { TRANSLATION_CONFIG } from "./translation-config.constants";
+
+interface TranslationConfig {
+    service: TranslationServiceInterface;
+}
+
+@Global()
+@Module({})
+export class TranslationModule {
+    static register(config: TranslationConfig): DynamicModule {
```
Comment (johnnyomair) [inline packages/api/cms-api/src/translation/translation.module.ts:14]: This needs to support asynchronous registration. The translation service in the application shouldn't be provided by this module, but by a module in the application. For instance, there could be a `DeepLModule` that provides the service. This translation module would then need to import the module and use the service that's exported there.

File: packages/api/cms-api/src/translation/translation.module.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+import { TranslationServiceInterface } from "./translation.config";
+import { TranslationResolver } from "./translation.resolver";
+import { TRANSLATION_CONFIG } from "./translation-config.constants";
+
+interface TranslationConfig {
+    service: TranslationServiceInterface;
+}
+
+@Global()
```
Comment (johnnyomair) [inline packages/api/cms-api/src/translation/translation.module.ts:11]:
```suggestion
```

I don't think this is needed

File: packages/api/cms-api/src/translation/translation.module.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+@Module({})
+export class TranslationModule {
```
Comment (johnnyomair) [inline packages/api/cms-api/src/translation/translation.module.ts:13]: Please rename to `ContentTranslationModule` to be in line with the naming in the Admin

File: packages/api/cms-api/src/translation/translation.resolver.ts [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+
+import { TranslationConfig } from "./translation.config";
+import { TRANSLATION_CONFIG } from "./translation-config.constants";
+
+@Resolver(() => String)
+export class TranslationResolver {
+    constructor(@Inject(TRANSLATION_CONFIG) private readonly config: TranslationConfig) {}
+
+    @Query(() => String)
+    async translate(
```
Comment (johnnyomair) [inline packages/api/cms-api/src/translation/translation.resolver.ts:12]: I'd change this to `contentTranslationTranslate(text: String!, targetLanguage: String!)`:

- `contentTranslation` as "module prefix"
- `text` we're translating text
- `targetLanguage`: `language` is too unspecific. Is it the source language or the target language?

Come to think of it, why do we even need this module in the library? It doesn't do anything than call the service provided by the application -> we could implement it in the application

Comment (VP-DS) [reply packages/api/cms-api/src/translation/translation.resolver.ts]: Removed the api module implementation with an example in the demo: 79838339a24b83de6a330fa352cf78e017fe226f

File: packages/api/cms-api/src/translation/translation.module.ts (hunk 3 above)
Comment (VP-DS) [reply packages/api/cms-api/src/translation/translation.module.ts]: Resolved outdated.

File: packages/api/cms-api/src/translation/translation.module.ts (hunk 2 above)
Comment (VP-DS) [reply packages/api/cms-api/src/translation/translation.module.ts]: Resolved outdated.

File: packages/api/cms-api/src/translation/translation.module.ts (hunk 1 above)
Comment (VP-DS) [reply packages/api/cms-api/src/translation/translation.module.ts]: Resolved outdated.

File: packages/api/cms-api/src/translation/translation.module.ts [hunk 5]
```diff
@@ -0,0 +1,26 @@
+import { DynamicModule, Global, Module } from "@nestjs/common";
+
+import { TranslationServiceInterface } from "./translation.config";
+import { TranslationResolver } from "./translation.resolver";
+import { TRANSLATION_CONFIG } from "./translation-config.constants";
+
+interface TranslationConfig {
+    service: TranslationServiceInterface;
```
[the remark this answers is not in the document]
Comment (VP-DS) [reply packages/api/cms-api/src/translation/translation.module.ts]: Resolved outdated.

File: pnpm-lock.yaml [hunk 6]
```diff
@@ -817,25 +817,25 @@ importers:
         version: 3.7.4(graphql@15.8.0)(react-dom@17.0.2)(react@17.0.2)
       '@comet/admin':
         specifier: '*'
-        version: link:../packages/admin/admin
+        version: 5.5.0(@apollo/client@3.7.4)(@emotion/react@11.9.3)(@emotion/styled@11.10.5)(@mui/icons-material@5.11.0)(@mui/material@5.11.6)(@mui/styles@5.11.2)(@mui/x-data-grid@5.17.20)(@types/react@17.0.53)(final-form@4.20.9)(graphql@15.8.0)(history@4.10.1)(react-dnd@16.0.1)(react-dom@17.0.2)(react-final-form@6.5.9)(react-intl@5.25.1)(react-router-dom@5.3.4)(react-router@5.3.4)(react@17.0.2)
```
[the remark this answers is not in the document]
Comment (VP-DS) [reply pnpm-lock.yaml:820]: Resolved outdated.

File: packages/api/cms-api/src/translation/translation.config.ts [hunk 7]
```diff
@@ -0,0 +1,7 @@
+export interface TranslationServiceInterface {
+    translate(value: string, language: string): Promise<string>;
+}
+
+export interface TranslationConfig {
+    service: TranslationServiceInterface;
+}
```
[the remark this answers is not in the document]
Comment (VP-DS) [reply packages/api/cms-api/src/translation/translation.config.ts]: Resolved outdated.

File: demo/admin/src/translator/CustomRteField.tsx [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+                icon: Highlight,
+                style: {
+                    backgroundColor: "yellow",
+                },
+            },
+        },
+    },
+});
+
+export const customRteFieldInitalValue =
```
Comment (johnnyomair) [inline demo/admin/src/translator/CustomRteField.tsx:38]: Please remove this, this isn't necessary in Demo

File: demo/admin/src/common/MasterMenu.tsx [hunk 9]
```diff
@@ -89,6 +90,13 @@ export const masterMenuData: MasterMenuData = [
                     component: NewsPage,
                 },
             },
+            {
+                primary: <FormattedMessage id="menu.translator" defaultMessage="Translator" />,
+                route: {
+                    path: "/structured-content/translator",
+                    component: TranslatorPage,
+                },
+            },
```
Comment (johnnyomair) [inline demo/admin/src/common/MasterMenu.tsx:99]: I'd prefer if you'd demo the translation feature in the page tree, for instance `EditPage`

File: demo/api/src/auth/permission.interface.ts [hunk 10]
```diff
@@ -2,6 +2,7 @@ declare module "@comet/cms-api" {
     interface Permission {
         news: string;
         products: string;
+        translation: string;
```
Comment (johnnyomair) [inline demo/api/src/auth/permission.interface.ts:5]: Note: This change will be unnecessary with #1721

**Comment by johnnyomair** [comment]
Do we still need this when we merge #1986?

**Comment by VP-DS** [comment]
> Do we still need this when we merge #1986?

I guess not.

**Pull request closed without merging**